src/main/java/maksym/fedorenko/bookstore/model/ShoppingCart.java

```diff
@@ -0,0 +1,42 @@
+package maksym.fedorenko.bookstore.model;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.MapsId;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Table(name = "shopping_carts")
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted=false")
```

Comment:
do you really need soft delete for shopping carts? cart items does not support soft delete, which means you'll end up with empty shopping carts left in the DB

also since you're using user id as the cart id - what will happen when a cart is created again? will you set `isDeleted` to true?

Reply:
fixed in the next PR